src/noit_metric.h

```diff
@@ -77,9 +77,23 @@ typedef enum {
 } noit_message_type;
 
 typedef struct {
+  uint16_t total_size;
+  uint8_t category_size;
+  const char *tag;
+} noit_metric_tag_t;
+
+typedef struct noit_metric_tagset_t {
+  noit_metric_tag_t *tags;
```

Comment:
life-cycle management of this structure is complicated, since `tags` point into user-supplied memory, so the source memory must outlive this structure...

Reply:
That's why I dropped the forced canonicalization.. The caller here already has memory it is ensuring that will last longer than the tagset, so we'd like to avoid the addition allocation.